ksqldb-rest-app/src/test/java/io/confluent/ksql/rest/integration/PullQueryRoutingFunctionalTest.java

```diff
@@ -258,19 +249,13 @@ public void cleanUp() {
     APP_SHUTOFFS_2.reset();
   }
 
-  @AfterClass
-  public static void classTearDown() {
-    TMP.delete();
-  }
-
   @Test
   public void shouldQueryActiveWhenActiveAliveQueryIssuedToStandby() throws Exception {
     // Given:
-    ClusterFormation clusterFormation = findClusterFormation(REST_APP_0, REST_APP_1, REST_APP_2);
+    ClusterFormation clusterFormation = findClusterFormation(TEST_APP_0, TEST_APP_1, TEST_APP_2);
     waitForClusterToBeDiscovered(clusterFormation.standBy.getApp(), 3);
     waitForRemoteServerToChangeStatus(clusterFormation.router.getApp(),
-        clusterFormation.router.getHost(),
-        PullQueryRoutingFunctionalTest::lagsExist);
+        clusterFormation.router.getHost(), HighAvailabilityTestUtil.lagsExist(3));
```

Comment:
Why do we need the lagsExists check here?

Reply:
The intention is to ensure that all lags have been reported since we won't forward requests if we have no lag info by default.  If we sneak in a pull query before this happens, it will fail, so we just wait for all lags to be reported.